biodatasets/pubtator_central/pubtator_central.py

```diff
@@ -0,0 +1,322 @@
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="pubtator_central_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central source schema",
+            schema="source",
+            subset_id="pubtator_central",
+        ),
+        BigBioConfig(
+            name="pubtator_central_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central",
+        # sample source 
+        BigBioConfig(
+            name="pubtator_central_sample_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central sample source schema",
+            schema="source",
+            subset_id="pubtator_central_sample",
+        ),
+        # sample big bio
+        BigBioConfig(
+            name="pubtator_central_sample_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central sample BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central_sample",
+        ),
+        # full dataset source
+        BigBioConfig(
+            name="pubtator_central_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central source schema",
+            schema="source",
+            subset_id="pubtator_central",
+        ),
+        # full dataset bigbio
+        BigBioConfig(
+            name="pubtator_central_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "pubtator_central_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "pmid": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "abstract": datasets.Value("string"),
+                    "mentions": [
+                        {
+                            "concept_id": datasets.Value("string"),
+                            "type": datasets.Value("string"),
+                            "text": datasets.Value("string"),
+                            "offsets": datasets.Sequence(datasets.Value("int32")),
+                        }
+                    ],
+                }
+            )
+
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
```

Comment:
here we will need to select the full dataset or sample dataset (probably using `self.config.subset_id`)

Reply:
Done :+1: